src/models/blueprint-collection.js

```diff
@@ -0,0 +1,105 @@
+import path from 'path';
+import fs from 'fs';
+import _map from 'lodash/map';
+import _filter from 'lodash/filter';
+import _isNil from 'lodash/isNil';
+import _isBool from 'lodash/isBoolean';
+import _isString from 'lodash/isString';
+import _isArray from 'lodash/isArray';
+import _flatten from 'lodash/flatten';
+import _uniq from 'lodash/uniq';
+import Blueprint from './blueprint';
+
+export default class BlueprintCollection {
+  constructor (pathList) {
+    this.pathList = pathList;
+    this.setSearchPaths();
+  }
+
+  allPossiblePaths () {
+    return _flatten(
+      _map(
+        this.pathList,
+        (arr, base) => _map(arr, (bp) => expandPath(base, bp) ))
+    );
+  }
+
+  setSearchPaths () {
+    this.searchPaths = _uniq(_filter(this.allPossiblePaths(), validSearchDir));
+  }
+
+  all() {
+    // Is there a more idiomatic way to do this?  I miss ruby's ||=
+    if (this.allBlueprints) {
+      return this.allBlueprints;
+    } else {
+      return this.allBlueprints = this.discoverBlueprints();
+    }
+  }
+
+  generators() {
+    // until we learn to tell generators apart from partials
+    return _filter(this.all(), (bp) => bp.name);
+  }
+
+  allNames() {
+    return _map(this.all(), (bp) => bp.name);
+  }
+
+
+  addBlueprint(path) {
+    return Blueprint.load(path);
+  }
+
+  discoverBlueprints(){
+    return _map(this.findBlueprints(), this.addBlueprint);
+  }
+
+  findBlueprints() {
+    return _flatten(
+      _map(
+        this.searchPaths,
+        (dir) => {
+          const subdirs = _map(fs.readdirSync(dir), (p) => path.resolve(dir, p));
+          return _filter(subdirs, (d)=> fs.existsSync(path.resolve(d, 'index.js')));
+        }
+      )
+    );
+  }
+}
+
+function validSearchDir(dir) {
+  return fs.existsSync(dir) && fs.lstatSync(dir).isDirectory();
+}
+
+export function expandPath (base, candidate) {
+  let final;
+  if (candidate[0] === '~') {
+    const st = candidate[1] === path.sep ? 2 : 1;
+    final = path.resolve(process.env.HOME, candidate.slice(st));
+  } else if (candidate[0] === path.sep) {
+    final = path.resolve(candidate);
+    // } else if (candidate[0] === '@') {
+    //   return path.join(npmPath,npm name, 'blueprints');
+  } else {
+    final = path.resolve(base, candidate);
+  }
+  return final;
+}
+
+export function parseSetting (setting) {
```

Comment:
I can't see where parseSetting is used?

Reply:
yeah...

So that function is supposed to take the list of blueprint paths for a given .blueprintrc file and add the ./blueprints directory to the end of that list.

I believe I wrote it and then became unsure of where it should go.  Let me fix that.

Reply:
Well it is actually being used.

it's imported in `models/project-settings` and had it's name changed to `parseBlueprint`. which is used in the Rc parser wrapper.  It's job is to turn whatever the settings are into arrays, and to add `./blueprints` to the end of those arrays.  this lets you do 

It's defined in BlueprintCollection because that's the concerned party
It's called in ProjectSettings because that's where that job needs to place

I changed the name to parseBlueprintSetting for both usages.

Reply:
If you're ok with this, I'm going to merge